Sources/Apollo/HTTPNetworkTransport.swift

```diff
@@ -258,15 +319,31 @@ public class HTTPNetworkTransport: NetworkTransport {
     return request
   }
 
-  private func requestBody<Operation: GraphQLOperation>(for operation: Operation) -> GraphQLMap {
-    if sendOperationIdentifiers {
+  private func requestBody<Operation: GraphQLOperation>(for operation: Operation,
+                                                        sendQueryDocument: Bool,
+                                                        autoPersistQueries: Bool) -> GraphQLMap {
+    
+    var payload: GraphQLMap = [:]
+    
+    if autoPersistQueries {
       guard let operationIdentifier = operation.operationIdentifier else {
```

Comment:
`--operationIdsPath` is path to JSON file that is tooling dependant and/or can be modified as you see fit

Reply:
The question here is Do we need to support autoPersistQueries just for some request but not for all?
As the autoPersistQueries seems network traffic optimisation like CDN caching which should be applying to all or none because it depends server side ability to support or not.

I just worry if we don't `preconditionFailure` for the pre-gen code that's not fulfilling the APQs requirement (but dev opt to turn on APQs). Developers will not be able to get alert about the run script missing `operationIdsPath` until they look into the traffic logging and find out APQs isn't in place.

In addition, I think the JSON file will not affect the code gen? I mean even you modify the JSON file, it won't affect operation.operationIdentifier in the generated Code?

Reply:
IIRC the operation identifier is based on the hash of the body, so it shouldn't change based on the location of the codegen. 

Personally I'm good with leaving this as a precondition failure - @codebear98 is correct, if someone wants APQ's and hasn't set up generating operation IDs, we want this to fail loud, hard, and early.

Reply:
fair enough